packages/integration-tests/src/test-e2e/deployment.test.ts

```diff
@@ -17,155 +21,175 @@ import {
 import assert from 'node:assert';
 import { TestBranch, amplifyAppPool } from '../amplify_app_pool.js';
 import { BackendIdentifier } from '@aws-amplify/plugin-types';
+import { testConcurrencyLevel } from './test_concurrency.js';
+
+const testProjectCreators = getTestProjectCreators();
+void describe(
+  'amplify deploys',
+  { concurrency: testConcurrencyLevel },
+  async () => {
+    before(async () => {
+      await createTestDirectory(rootTestDir);
+    });
+    after(async () => {
+      await deleteTestDirectory(rootTestDir);
+    });
 
-const testProjects = await generateTestProjects(rootTestDir);
-
-void describe('amplify deploys', async () => {
-  after(async () => {
-    await deleteTestDirectory(rootTestDir);
-  });
-
-  testProjects.forEach((testProject) => {
-    void describe(`branch deploys ${testProject.name}`, () => {
-      let branchBackendIdentifier: BackendIdentifier;
-      let testBranch: TestBranch;
-
-      beforeEach(async () => {
-        testBranch = await amplifyAppPool.createTestBranch();
-        branchBackendIdentifier = {
-          namespace: testBranch.appId,
-          name: testBranch.branchName,
-          type: 'branch',
-        };
-      });
-
-      afterEach(async () => {
-        await testProject.tearDown(branchBackendIdentifier);
-      });
-
-      void it(`[${testProject.name}] deploys fully`, async () => {
-        await testProject.deploy(branchBackendIdentifier);
-        await testProject.assertPostDeployment();
-        const testBranchDetails = await amplifyAppPool.fetchTestBranchDetails(
-          testBranch
-        );
-        assert.ok(
-          testBranchDetails.backend?.stackArn,
-          'branch should have stack associated'
-        );
-        assert.ok(
-          testBranchDetails.backend?.stackArn?.includes(
-            branchBackendIdentifier.namespace
-          )
-        );
-        assert.ok(
-          testBranchDetails.backend?.stackArn?.includes(
-            branchBackendIdentifier.name
-          )
-        );
+    testProjectCreators.forEach((testProjectCreator) => {
+      void describe(`branch deploys ${testProjectCreator.name}`, () => {
+        let branchBackendIdentifier: BackendIdentifier;
+        let testBranch: TestBranch;
+        let testProject: TestProjectBase;
+
+        beforeEach(async () => {
+          testProject = await testProjectCreator.createProject(rootTestDir);
+          testBranch = await amplifyAppPool.createTestBranch();
+          branchBackendIdentifier = {
+            namespace: testBranch.appId,
+            name: testBranch.branchName,
+            type: 'branch',
+          };
+        });
+
+        afterEach(async () => {
+          await testProject.tearDown(branchBackendIdentifier);
+        });
+
+        void it(`[${testProjectCreator.name}] deploys fully`, async () => {
```

Comment:
nit: having project name here and on line 39 is repetitive

Reply:
same for sandbox, but it doesn't seem to produce bad output, so I skipped this.